File: frontend/src/app/services/customer.service.ts
import {Observable} from "rxjs";
import {customerModel} from "../modules/models/customerModel";
import {customerOrErrorsModel} from "../modules/models/customerOrErrorsModel";
import {CustomerPageModel} from "../modules/models/CustomerPageModel";

export interface CustomerService {
  checkAndSaveCustomer(customer: customerModel): Observable<customerOrErrorsModel>;

  findCustomerByLogInInfId(logInInfId: number): Observable<customerModel>;

  updateCustomerPersonalInf(customer: customerModel): Observable<customerOrErrorsModel>;

  saveCustomerWallet(customer: customerModel): Observable<customerOrErrorsModel>;

  findCustomerById(id: number): Observable<customerModel>;

  findAllByPage(pageNumber: number, amount: number): Observable<CustomerPageModel>;

  changeStatus(customerModel: customerModel): Observable<customerOrErrorsModel>;

  liquidateDebt(customerModel: customerModel): Observable<customerOrErrorsModel>;
}

File: be/src/main/java/com/artsykov/backend/service/CustomerService.java
package com.artsykov.backend.service;

import com.artsykov.backend.entity.CustomerEntity;
import com.artsykov.backend.model.CustomerPageModel;

public interface CustomerService {
    CustomerEntity saveCustomer(CustomerEntity customer);

    CustomerEntity findCustomerByLogInInfId(int logInInfId);

    void updateCustomer(CustomerEntity customerEntity);

    CustomerEntity saveCustomerWallet(CustomerEntity customerEntity);

    CustomerEntity findCustomerById(int customerId);

    CustomerPageModel findAllByPage(int pageNumber, int amount);

    void changeStatus(CustomerEntity customerEntity);

    CustomerEntity liquidateDebt(CustomerEntity customerEntity);
}
